src/lib.rs

```diff
@@ -67,7 +67,7 @@ pub mod registers;
 pub mod structures;
 
 /// Represents a protection ring level.
-#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
```

Comment:
I'm not sure about the `PartialOrd`/`Ord` derives: When talking about ring 0, ring 3, etc, it makes sense that `Ring3` is considered larger than `Ring0`. However, the name of the type — `PrivilegeLevel` — kind of indicates the opposite ordering: One might reasonably expect that the highest privilege level has the most privileges, i.e. that `Ring0` is the highest level.

Perhaps it's better to not derive `PartialOrd` here in order to prevent confusion? Users can still introduce their own wrapper types with an `PartialOrd` implementation if they want.